Security docs update (#680)
OctopusDeploy/docs: docs/administration/security/hardening-octopus.md

|WinRM-HTTP|`TCP 5985`|Octopus Server|Anywhere|DENY|Prevents attackers from using the Octopus Server as a beachhead into your network via unsecured WinRM.|
|All outbound|`ALL`|Octopus Server|Anywhere|???|Depends on how fine-grained control you want over what your Octopus Server can do. It also depends on where your workers are, and where you are deploying to. Allowing all outbound traffic is a good place to start, then perform network analysis to decide on your next step.|

## Harden your containers

If you run an [Octopus Deploy container](/docs/installation/octopus-in-container/index.md), in addition to your usual security measure for running apps out of containers, take the following steps to secure it: 

> - Store your containers on a separate partition.

Reviewer: Are you talking about the volumes? the images?
_I'm not even sure how to do this..._ (not that I'm anywhere near an expert)

Reviewer 2: I reckon that advice (assume it comes from here: https://www.secjuice.com/how-to-harden-docker-containers/) has got to be referring to the docker data directory that lives in `/var/lib/docker`. By moving this to a different partition, it makes it harder for a compromised container to then reach into other places in `/var` (I think)

Reviewer 2: ![image](https://user-images.githubusercontent.com/579657/79316730-e59c8d00-7f43-11ea-8852-66d3e65413cd.png)

Reviewer 2: How to move that folder:
https://www.guguweb.com/2019/02/07/how-to-move-docker-data-directory-to-another-location-on-ubuntu/

- Assign resources carefully:
  - Consider pinning CPUs to namespaces in order to give them a boundary. 

>   - Consider the amount of memory required, too much and the container is susceptible to denial of service attacks, too little or the use of ballooning will hinder performance. 

Reviewer 3: Language tweaks. 
```suggestion
  - Consider the amount of memory required, if you assign too much the container is susceptible to denial of service attacks, but if you assign too little or the use of ballooning will hinder performance. 
```

> - Network namespaces - consider which containers reside in each network namespace as all processes in a namespace can talk to the namespace interface.

Reviewer 3: ```suggestion
- Consider which containers reside in each network namespace as all processes in a namespace can talk to the namespace interface.
```

> The security of your containers and Docker configuration can be analyzed in detail by using [Docker Bench for Security](https://github.com/docker/docker-bench-security) from the [Center for Internet Security](https://www.cisecurity.org/about-us/). 

Reviewer: AFAIK, this only works for linux containers.
Also, not sure how we rate on this. Might be worth checking that out first.

Reviewer 2: I'm taking a look at this now.

Reviewer 2: Looks the purpose of this tool is to check the _Docker hosts_ security, whereas the above advice seems to indicate that it will check 'inside' your containers. @matt-richardson is correct insofar that this will only work on Linux hosts. We don't need to check ourselves against this benchmark in hosted because MS is responsible for the security of the underlying infrastructure that runs our k8s clusters.
That all said, I'd reword this slightly:
```suggestion
The security of your container host and its Docker configuration can be analyzed in detail by using [Docker Bench for Security](https://github.com/docker/docker-bench-security) from the [Center for Internet Security](https://www.cisecurity.org/about-us/). 
```

Reviewer 2: _pretty neat tool though I ran int on my machine and it got me thinking about some stuff :)_

That tool is inspired by this organizations work, which looks like it is much less platform specific: https://www.cisecurity.org/benchmark/docker/ and could be used by a windows server admin to benchmark their own VMs/AMIs/ARMs

If you're interested in seeing the details of that benchmark I've pinned it in #school-security here https://octopusdeploy.slack.com/archives/CALSGJK7Z/p1586941721051700

## Samples

> We have an [Octopus Admin](https://g.octopushq.com/OctopusAdminSamplesSpace) Space on our Samples instance of Octopus.  Here you can sign in as `Guest` and take a look at some examples where we have used Octopus for hardening tasks. 

Reviewer 3: ```suggestion
We have an [Octopus Admin](https://g.octopushq.com/OctopusAdminSamplesSpace) Space on our Samples instance of Octopus. You can sign in as `Guest` to take a look at some examples of how we have used Octopus for hardening tasks. 
```

|WinRM-HTTP|`TCP 5985`|Octopus Server|Anywhere|DENY|Prevents attackers from using the Octopus Server as a beachhead into your network via unsecured WinRM.|
|All outbound|`ALL`|Octopus Server|Anywhere|???|Depends on how fine-grained control you want over what your Octopus Server can do. It also depends on where your workers are, and where you are deploying to. Allowing all outbound traffic is a good place to start, then perform network analysis to decide on your next step.|

## Harden your containers

> If you run an [Octopus Deploy container](/docs/installation/octopus-in-container/index.md) these are a few of the steps you can take to secure it. 

Reviewer 3: I'd make the language a bit more certain here:
```suggestion
If you run an [Octopus Deploy container](/docs/installation/octopus-in-container/index.md), you need to take the following steps to secure it: 
```
If we're worried about giving users the impression these are the only steps they need to take, we can `in addition to your usual security measure for running apps out of containers, you need to...

- Store your containers on a separate partition.

> - Assign resources carefully

Reviewer 3: ```suggestion
- Assign resources carefully:
```
